### PR TITLE
[HW11] Add `DestroyUser` mutation

### PR DESCRIPTION
**Pull request by Ragnoji**
### Summary

A brief description of the pull request.

### How it works

Screenshots/screencasts of the pull request introduced functionality.

### Test plan

List of steps to manually test introduced functionality:

- Go to http://localhost:3000/graphiql
- Make request using schema:
```graphql
  query {
    me {
      id
      name
    }
  }
```

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, services, scripts, etc.

### Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My pull request is shorter than 500 lines

### CODE REVIEW

File: app/graphql/mutations/destroy_user.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  class DestroyUser < BaseMutation
+    include GraphqlErrors
+
+    argument :input, Types::Inputs::DestroyUserInput, required: true
+
+    type Types::Payloads::DestroyUserPayload
+
+    def resolve(input:)
+      user = User.find(input.id)
+      return { errors: [{ message: "Not authorized", backtrace: [] }] } unless user.eql?(current_user) || user.admin?
```
Comment (RafaelKhammadeev) [inline app/graphql/mutations/destroy_user.rb:11]: I think it should be in Policies, but it's cool:D

Comment (RafaelKhammadeev) [reply app/graphql/mutations/destroy_user.rb:11]: Why you use backtrace in errors?

**Review by RafaelKhammadeev — COMMENTED** [review]
Good job bro:D

File: app/graphql/types/user_type.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Types
+  class UserType < Types::BaseObject
+    field :id, ID
+    field :first_name, String
+    field :last_name, String
+    field :email, String
+    field :role, String
```
Comment (RasimKhusaenov) [inline app/graphql/types/user_type.rb:7]: It is okay, but ideally it should be special type:
```ruby
module Types
  class UserRoleEnum < BaseEnum
    User.role.each_value do |user_role|
      value user_role.to_s.upcase, user_role.text, value: user_role.to_s
    end
  end
end
```

Comment (Ragnoji) [reply app/graphql/types/user_type.rb]: Thanks

File: app/graphql/mutations/destroy_user.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module Mutations
+  class DestroyUser < BaseMutation
+    include GraphqlErrors
+
+    argument :input, Types::Inputs::DestroyUserInput, required: true
```
Comment (RasimKhusaenov) [inline app/graphql/mutations/destroy_user.rb:5]: How about refactor it for destroying current user account?

**Review by RasimKhusaenov — CHANGES_REQUESTED** [review]

File: app/graphql/types/user_role_type.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module Types
+  class UserRoleEnum < BaseEnum
```
Comment (RasimKhusaenov) [inline app/graphql/types/user_role_type.rb:2]: Wrong filename!

File: app/interactors/users/destroy.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module Users
+  class Destroy
+    include Interactor
+
+    delegate :user, to: :context, allow_nil: false
```
Comment (RasimKhusaenov) [inline app/interactors/users/destroy.rb:5]: `allow_nil: false` doesn't work as expected

**Review by RasimKhusaenov — CHANGES_REQUESTED** [review]

**Review by RasimKhusaenov — APPROVED** [review]

**Pull request merged**